db/package.js

```diff
@@ -102,61 +102,96 @@ class PackageDbController {
+
+    // of the packages already marked as maintained by me, whichever aren't in the list of
+    // packages provided, remove my id from their maintainers list
     const packageDeletions = existingPackages
       .filter(pkg => !packages.includes(pkg.name))
       .map(pkg => ({
-        criteria: { name: pkg.name, registry },
+        criteria: { _id: pkg._id },
         update: {
-          $set: { owner: null },
           $pull: { maintainers: { maintainerId } }
         }
       }))
+
+    // of the packages provided, whichever aren't already marked as maintained by me,
+    // push my id to their maintainers list (upsert if the package isn't in the DB at all)
     const packageInsertions = packages
       .filter(pkg => !existingPackages.some((ePkg) => ePkg.name === pkg))
       .map(pkg => ({
-        name: pkg,
-        registry,
-        owner: maintainerId,
-        maintainers: [{ maintainerId, revenuePercent: 100 }]
+        criteria: {
+          name: pkg,
+          registry
+        },
+        update: {
+          $push: {
+            maintainers: { maintainerId, revenuePercent: 0 }
+          }
+        }
       }))
+
+    // of the packages found in the DB, make sure I am marked as a maintainer
+    // of the ones I maintain
     const packageUpdates = existingPackages
-      .filter(pkg => pkg.owner !== maintainerId)
-      .map(pkg => {
-        const alreadyMaintains = pkg.maintainers.some(maintainer => maintainer.maintainerId === maintainerId)
-        return {
-          criteria: { name: pkg.name, registry },
-          update: {
-            $set: {
-              owner: maintainerId,
-              maintainers: alreadyMaintains
-                ? pkg.maintainers
-                : pkg.maintainers.concat([{ maintainerId, revenuePercent: 0 }])
+      .filter(pkg => !pkg.maintainers || !pkg.maintainers.some(m => m.maintainerId === maintainerId))
+      .map(pkg => ({
+        criteria: { _id: pkg._id },
+        update: {
+          $push: {
+            maintainers: {
+              maintainerId,
+              revenuePercent: 0
             }
           }
         }
-      })
+      }))
+
     const bulkPackages = this.db.collection('packages').initializeUnorderedBulkOp()
 
     for (const insertion of packageInsertions) {
-      bulkPackages.insert(insertion)
-    }
-    for (const update of packageUpdates) {
-      bulkPackages.find(update.criteria).update(update.update)
+      bulkPackages.find(insertion.criteria).upsert().update(insertion.update)
     }
     for (const deletion of packageDeletions) {
       bulkPackages.find(deletion.criteria).update(deletion.update)
     }
+    for (const update of packageUpdates) {
+      bulkPackages.find(update.criteria).update(update.update)
+    }
+
+    // due to how the insertions/updates were done, there may be some packages that were upserted
+    // or updated and have only one maintainer with 0 percent revenue share. to fix that, we'll
+    // change those to 100%
```

Comment:
wondering if we can do this when we "create new packages that weren't there before" up on line 131

Reply:
updated L131 but i think there are still the case of 

a) package that i used to own, i don't own anymore. there were 2 maintainers and i just got popped off. now the remaining maintainer needs 100%

Reply:
great point